patroni/dcs/consul.py

```diff
@@ -324,12 +346,52 @@ def touch_member(self, data, ttl=None, permanent=False):
         try:
             args = {} if permanent else {'acquire': self._session}
             self._client.kv.put(self.member_path, json.dumps(data, separators=(',', ':')), **args)
+            if self._register_service:
+                self.update_service(self._my_member_data, data)
             self._my_member_data = data
             return True
         except Exception:
             logger.exception('touch_member')
         return False
 
+    @catch_consul_errors
+    def register_service(self, service_name, **kwargs):
+        logger.info('Register service %s, params %s', service_name, kwargs)
+        return self.retry(self._client.agent.service.register, service_name, **kwargs)
+
+    def _update_service(self, data):
+        service_name = self._service_name
+        role = data['role']
+        api_parts = urlparse(data['api_url'])
+        api_parts = api_parts._replace(path='/{}'.format(role))
+        conn_parts = urlparse(data['conn_url'])
+        params = {
+            'address': conn_parts.hostname,
```

Comment:
We should also add ` 'service_id': service_name_from_scope_name(self._name),` to params, otherwise it is using `service_name` as a service_id, what makes impossible to run master and replica with the same agent.

Reply:
i've added deregistration and service_id is now based on scope_name and name